### ISSUE TITLE
Used --statistics option does hide the rule violations

### ISSUE THREAD

**Issue by ssbarnea** [astral-sh/ruff#14451]
The fact that `ruff check --statistics` does disabled the output of effective line violations makes it impossible to us this as an addon option for improved output, especially when using ruff using as a pre-commit hook.

If that was its initial intention, it was implemented wrongly because it should have being just `--output-format=statistics` if it replaces the output.

This could be sorted by ensuring that `--statistics` does not affect the output format and that is output goes to stderr so it would not break tools that might process the standard output formats, which could be machine parseable.

Story: As an user, I want to see some stats about found violations, after they are listed.

**Comment by MichaReiser** [comment]
I understand your use case. But just writing to stderr is not necessarily a solution because `--statistics` also impacts other output formats. 

```
❯ uvx ruff check . --statistics --output-format json > output.json

test on  main [$✘!?] via 🐍 v3.12.7 
❯ bat output.json
───────┬───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: output.json
───────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ [
   2   │   {
   3   │     "code": "I001",
   4   │     "name": "unsorted-imports",
   5   │     "count": 6,
   6   │     "fixable": true
   7   │   }
   8   │ ]
```

I think the ask here is to make `--statistics` additive. Ruff should include the statistics in the output in addition to the individual diagnostics. But I'm not 100% sure about that. This requires fleshing out the UX first

**Comment by luisrayas3** [comment]
I would also like to see both, and given both comments perhaps @ssbarnea is right and it should be implemented as an output-format?
```sh
uvx ruff check --output-format statistics  # Current --statistics behavior
uvx ruff check --output-format full-with-statistics  # Additive
uvx ruff check --output-format json  # Cannot be mangled by additive statistics
```
I might be worried about proliferation of formats but considering most are programmatic you'd only really need to add `full-with-statistics` and `concise-with-statistics`
